### ISSUE TITLE
Use of assets volume is problematic, see explanation

### ISSUE THREAD

**Issue by rmmr** [frappe/frappe_docker#790]
The current solution you are providing in compose.yaml for sharing the build assets from the nginx container to the backend containers is problematic. Upon creating the volume, as stated in the  docker docs, it will populate the assets volume with the contents from the frontend container at `/usr/share/nginx/html`, this is fine. However:

Any new nginx images which have new assets at `/usr/share/nginx/html`, will not propogate these changes if the volume already exists. Thus in order to fully update the (production) environment the following wont work:

```
docker pull <name of nginx image>
docker-compose up -d
```

One has to:

```
docker pull <name of nginx image>
docker-compose down
docker volume rm <name of assets volume>
docker-compose up -d
```

Perhaps this has always been intended, but it would be good to document this. There are imo two solutions to truly fix this:

1. Use the nginx image as a multi stage source for the worker image and copy the contents from `/usr/share/nginx/html` to `/home/frappe/frappe-bench/sites/assets`. However this won't work due to the need for the `sites` volume. If the assets folder can be moved outside the sites folder directly under frappe-bench, this would work. However a change in frappe itself is needed.

2. Have the frontend image mount the `assets` volume under `/tmp/assets` for example. Then have an entrypoint script copy over `/usr/share/nginx/html` to `/tmp/assets`, thus overwriting the volume contents.

**Comment by revant** [comment]
> 2\. Then have an entrypoint script copy over `/usr/share/nginx/html` to `/tmp/assets`, thus overwriting the volume contents.

this is what I'm doing in helm charts

https://github.com/frappe/helm/blob/91643491bbbe19313d9b456d2648490fb43fac9e/erpnext/templates/deployment-gunicorn.yaml#L34-L39

**Comment by revant** [comment]
another idea is to add one more service in docker-compose.yml that does this and successfully completes.

**Comment by revant** [comment]
> One has to:
> 
> ```
> docker pull <name of nginx image>
> docker-compose down
> docker volume rm <name of assets volume>
> docker-compose up -d
> ```

Documented this in #791

**Comment by rmmr** [comment]
Perhaps the best approach in the long run would be to file an issue at frappe/frappe, to support solution 1. Usage of containers, should be simple, reproducible, and most importantly the build time dependencies should be "CONTAINED".

**Comment by revant** [comment]
That's a can of worms. I'm not going to open.

I gave up as soon as I found out node-sass is part of production dependencies. Forget "CONTAINED", It dynamically generates css when you save `Website Theme`.

my strategy is to just make it work containerized by doing whatever it takes, I try to be closer to ideal containers. that's why I'm treating assets as cache.

Upstream, anything that has to do with containers won't be a priority. The only choice we have as community is to containerize whatever is available.
